### PR TITLE
Integrate router into react state. Resolves GH-1689

### PR DESCRIPTION
**Pull request by thisandagain**

### CODE REVIEW

**Comment by thisandagain** [comment]
@k88hudson R?

File: www_src/lib/router.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      // Check to see if route params exist & create cache key
+      var routeParams = this.android.getRouteParams();
+      var key = 'state::route';
+
+      // If they do, save them. If not, fetch from SharedPreferences
+      if (routeParams !== '{}') {
+        r = JSON.parse(routeParams);
+        this.android.setSharedPreferences(key, routeParams, true);
+      } else {
+        var hit = this.android.getSharedPreferences(key, true);
```
Comment (k88hudson) [inline www_src/lib/router.jsx]: i cant remember, is it possible this could be `null`?

**Comment by k88hudson** [comment]
Ok, this looks good to me

File: www_src/lib/router.jsx (hunk 1 above)
Comment (thisandagain) [inline www_src/lib/router.jsx]: I couldn't think of how, but probably best to be safe than sorry.

**Pull request merged**